Update release process (#1268)
ezsystems/developer-documentation: docs/community_resources/release_process.md

[[= product_name =]] has three distributions:

- [[= product_name_content =]] is a multichannel and headless content management system.

> - [[= product_name_exp =]] is a modern modular Digital Experience Platform to build outstanding customer experiences

Reviewer: ```suggestion
- [[= product_name_exp =]] is a modern modular Digital Experience Platform for building outstanding customer experiences.
```

Author: This is actual phrasing from https://www.ibexa.co/products/ibexa-experience, copied here on purpose.

Reviewer: Do you agree with the phrasing? (Something's off with it for me.) If you do, that's ok, let's just fix the fullstop.

- [[= product_name_com =]] is a commerce-ready B2B DXP designed to digitalize your business from customer awareness to purchase and beyond.

> Additionally, [[= product_name =]] also has an open-source version, Ibexa OSS,

Reviewer: ```suggestion
Additionally, [[= product_name =]] also has an open-source version called Ibexa OSS.
```

> developed by Ibexa together with the open source community.

Reviewer: ```suggestion
Ibexa OSS is developed by Ibexa together with the open source community.
```

## Versioning conventions

All [[= product_name =]] editions use [semantic versioning](http://semver.org/).  

The version number of [[= product_name =]] and all its internal components follows the semantic versioning conventions: vX.Y.Z.

- Changes to X indicate breaking changes. They usually concern mostly internal things, but developers should check in our change logs if they need to adjust their code to continue using the API or features. If there are larger breaks, this is announced well in advance of the upcoming release.
- Y represents new features and functionalities.
- Z represents patches, bug fixes, smaller improvements, etc.

Distribution files of our three editions are as follows:

> - for [[= product_name_content =]]: ezplatform-vX.Y.Z.tgz

Reviewer 2: We will change the name of the tarball as well. I guess it will be `ibexa-content-vX.Y.Z.tgz`

Reviewer 2: ```suggestion
- for [[= product_name_content =]]: ibexa-content-vX.Y.Z.tgz
```

> - for [[= product_name_exp =]]: ezplatformenterprise-vX.Y.Z.tgz

Reviewer 2: ```suggestion
- for [[= product_name_exp =]]: ibexa-experience-vX.Y.Z.tgz
```

> - for [[= product_name_com =]]: 

Reviewer 2: ```suggestion
- for [[= product_name_com =]]: ibexa-commerce-vX.Y.Z.tgz
```
